src/roiextractors/extractors/femtonicsimagingextractor/femtonicsimagingextractor.py

```diff
@@ -0,0 +1,343 @@
+        return self._video.shape[1], self._video.shape[2]  # (height, width)
+
+    def get_num_samples(self) -> int:
+        """Get the number of samples (frames) in the video."""
+        return self._video.shape[0]
+
+    def get_channel_names(self) -> List[str]:
+        """Get the channel names."""
+        return [self._selected_channel_name]
+
+    def get_sampling_frequency(self) -> float:
+        """Get the sampling frequency in Hz."""
+        time_per_frame_ms = self._get_time_per_frame()
+        if time_per_frame_ms is None:
+            raise ValueError("Sampling frequency could not be determined from metadata.")
+        return 1000.0 / time_per_frame_ms
+
+    def _get_time_per_frame(self) -> Optional[float]:
+        """Get time per frame from metadata."""
+        session_key = f"MSession_{self._munit}"
+        munit_key = f"MUnit_{self._munit}"
+        attrs = dict(self._file[session_key][munit_key].attrs)
+        return attrs.get(
+            "ZAxisConversionConversionLinearScale"
+        )  # ZAxisConversionConversionLinearScale gives the duration of one frame in milliseconds.
+
+    def get_dtype(self) -> np.dtype:
+        """Get the data type of the video."""
+        return self._video.dtype
+
+    def get_series(self, start_sample: Optional[int] = None, end_sample: Optional[int] = None) -> np.ndarray:
+        """Get a series of frames."""
+        return self._video.lazy_slice[start_sample:end_sample, :, :].dsetread()
+
+    # Femtonics-specific getter methods
+    def _get_session_uuid(self):
+        """
+        Get the session UUID from the root attributes.
+
+        Returns
+        -------
+        uuid : array or value
+            The session UUID.
+        """
+        attrs = dict(self._file.attrs)
+        return attrs.get("Uuid")
+
+    def _get_pixel_size(self) -> Tuple[float, float]:
+        """Get pixel size in micrometers."""
+        session_key = f"MSession_{self._munit}"
+        munit_key = f"MUnit_{self._munit}"
+        attrs = dict(self._file[session_key][munit_key].attrs)
+
+        x_size = attrs.get("XAxisConversionConversionLinearScale", 1.0)
+        y_size = attrs.get("YAxisConversionConversionLinearScale", 1.0)
+        return x_size, y_size
+
+    def _get_image_shape_metadata(self) -> Tuple[int, int, int]:
+        """
+        Get the image shape metadata (X, Y, Z dimensions) from the measurement unit attributes.
+
+        Returns
+        -------
+        shape : tuple
+            (XDim, YDim, ZDim) as (num_columns, num_rows, num_frames)
+        """
+        session_key = f"MSession_{self._munit}"
+        munit_key = f"MUnit_{self._munit}"
+        munit_attrs = dict(self._file[session_key][munit_key].attrs)
+        x_dim = munit_attrs.get("XDim")
+        y_dim = munit_attrs.get("YDim")
+        z_dim = munit_attrs.get("ZDim")
+        return (x_dim, y_dim, z_dim)
+
+    def _get_measurement_date(self) -> Optional[datetime]:
```

Comment:
```suggestion
    def _get_session_start_time(self) -> Optional[datetime]:
```

What do you think of just using nwb centered terminology? the main purpose of this library is to convert data to NWB so I think it is a good idea if we use this terminology.